x-pack/solutions/observability/plugins/apm/server/routes/traces/get_unified_trace_span.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SpanDocument } from '@kbn/apm-types';
+import { existsQuery, termQuery, termsQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { unflattenKnownApmEventFields } from '@kbn/apm-data-access-plugin/server/utils';
+import { SPAN_ID, TRACE_ID, PROCESSOR_EVENT } from '../../../common/es_fields/apm';
+import type { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+
+export async function getUnifiedTraceSpan({
+  spanId,
+  traceId,
+  apmEventClient,
+}: {
+  spanId: string;
+  traceId: string;
+  apmEventClient: APMEventClient;
```

Comment:
As mentioned before, this could be reused. You could make the `spanId` optional and add another optional field named `fields`.

```suggestion
  spanId,
  traceId,
  apmEventClient,
  fields = ['*']
}: {
  spanId?: string;
  traceId: string;
  apmEventClient: APMEventClient;
  fields?: string[]
```